includes/class-wc-product-variable.php

```diff
@@ -67,8 +67,10 @@ public function check_stock_status() {
 		} elseif ( $this->backorders_allowed() || $this->get_stock_quantity() > get_option( 'woocommerce_notify_no_stock_amount' ) ) {
 			$set_child_stock_status = 'instock';
 		}
+		
+		$set_child_stock_status = apply_filters( 'woocommerce_product_check_stock_status', $set_child_stock_status, $this );
```

Comment:
I'm also confused with this filter... if the status is 'discontinued' how will it know? currently my logic tells me it can only tell if it's instock or outofstock.

basing on this example,
````
// Do nothing on stock status check for custom stock status
add_filter( 'woocommerce_product_check_stock_status', function( $status, $product ) {

    if ( $product->stock_status === 'discontinued' ) {
        return '';
    }

    return $status;

}, 10, 2 );
````
if `$product->stock_status` is `outofstock`, I can tell `$status` to be `outofstock`.
but if `$product->stock_status` is `discontinued` what's the expected value of `$status`?